src/test/java/org/candlepin/model/ConsumerCloudDataTest.java

```diff
@@ -466,15 +466,73 @@ public void testSetConsumer() {
         assertEquals(consumer, consumerCloudData.getConsumer());
     }
 
+    @Test
+    void testUpdateFromNull() {
+        ConsumerCloudData original = new ConsumerCloudData()
+            .setCloudProviderShortName("aws")
+            .setCloudAccountId("acc1")
+            .setCloudInstanceId("inst1")
+            .setCloudOfferingIds(List.of("off1", "off2"));
+
+        original.updateFrom(null);
+
+        assertEquals("aws", original.getCloudProviderShortName());
+        assertEquals("acc1", original.getCloudAccountId());
+        assertEquals("inst1", original.getCloudInstanceId());
+        assertEquals(List.of("off1", "off2"), original.getCloudOfferingIds());
+    }
+
+    @Test
+    void testUpdateFromUpdatesOnlyChangedFields() {
+        ConsumerCloudData original = new ConsumerCloudData()
+            .setCloudProviderShortName("aws")
+            .setCloudAccountId("acc1")
```

Comment:
We should also verify that cloud account ID and offering IDs get updated as expected when they are updated. Currently there are no tests that validate that these fields get updated

Reply:
done